extra/mariabackup/changed_page_bitmap.cc

```diff
@@ -195,7 +195,7 @@ log_online_read_bitmap_page(
 	ut_a(bitmap_file->offset % MODIFIED_PAGE_BLOCK_SIZE == 0);
 	success = os_file_read(IORequestRead,
 			       bitmap_file->file, page, bitmap_file->offset,
-			       MODIFIED_PAGE_BLOCK_SIZE);
+			       MODIFIED_PAGE_BLOCK_SIZE) == DB_SUCCESS;
```

Comment:
This case is not covered by tests. When I change it like this all tests from `mariabackup` still passes.
```diff
--- a/extra/mariabackup/changed_page_bitmap.cc
+++ b/extra/mariabackup/changed_page_bitmap.cc
@@ -195,7 +195,7 @@ log_online_read_bitmap_page(
        ut_a(bitmap_file->offset % MODIFIED_PAGE_BLOCK_SIZE == 0);
        success = os_file_read(IORequestRead,
                               bitmap_file->file, page, bitmap_file->offset,
-                              MODIFIED_PAGE_BLOCK_SIZE) == DB_SUCCESS;
+                              MODIFIED_PAGE_BLOCK_SIZE) != DB_SUCCESS;
 
        if (UNIV_UNLIKELY(!success)) {

```

Reply:
Changed-page tracking is an XtraDB extension that is not present in 10.2. So, this is dead code in 10.2+ until and unless we decide to implement the feature in MariaDB.